include/mbgl/style/conversion/get_json_type.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <string>
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+
+template <class V>
+std::string getJSONType(const V& value) {
+    using namespace mbgl::style::conversion;
+    
+    if (isUndefined(value)) {
+        return "null";
+    }
+    if (isArray(value)) {
+        return "array";
+    }
+    if (isObject(value)) {
+        return "object";
+    }
+    optional<mbgl::Value> v = toValue(value);
+    assert(v);
```

Comment:
This is based on the assumption that `V` is guaranteed to successfully represent one of the JSON value types -- since we've already checked the non-atomic types above it must then be a string, number, or boolean, so that `toValue()` should work.  I'm not sure what we'd do if that assumption weren't to hold...  but at the least, I can add a comment to justify the `assert`